Branch-SDK/src/io/branch/referral/PrefHelper.java

```diff
@@ -152,6 +157,7 @@ private PrefHelper(Context context) {
         this.prefsEditor_ = this.appSharedPrefs_.edit();
         this.requestMetadata = new JSONObject();
         this.installMetadata = new JSONObject();
+        this.secondaryRequestMetadata = new JSONObject();
```

Comment:
This is just a comment:

If`secondaryRequestMetadata` were null to start, you might be able to short circuit a whole ton of checks with a simple null check.

For example `getSecondaryRequestMetaData()` could check for null instead of doing the JSON lookup.  You would only need to create it in `addSecondaryRequestMetadata()` -- which again, is called very very seldom called.